whad/device/uart.py

```diff
@@ -89,7 +89,7 @@ def __init__(self, port='/dev/ttyUSB0', baudrate=115200):
         if port_info is None:
             raise WhadDeviceNotFound
         else:
-            self.__is_acm = (port_info.subsystem == 'usb')
+            self.__is_acm = (port_info.usb_info is not None)
```

Comment:
Linux systems differentiate a serial device exposing an interface through  USB CDC ACM from generic serial adapters like FTDIs by referencing them as using the "usb" subsystem (the other classic serial interfaces using the "usb-serial" subsystem). Checking that `usb_info` is not `None` might not be enough on Linux system to tell USB CDC ACM devices (usually detected as `/dev/ttyACMX` and FTDI adapters (usually detected as `/dev/ttyUSBX`) apart. 

```suggestion
            # 'subsystem' is only defined on Linux serial ports, we check if this property is missing to
            # handle Mac OS (Windows system is marked as unsupported for now)
            if not hasattr(port_info, "subsystem"):
                # Determine if device is ACM on Mac OS
                self.__is_acm = (port_info.usb_info is not None)
            else:
                # On Linux systems, ACM devices use the "usb" subsystem.
                self.__is_acm = (port_info.subsystem == "usb")
```

or a simpler form:

```suggestion
            self.__is_acm = (port_info.subsystem == "usb") if hasattr(port_info, "subsystem") else (port_info.usb_info is not None)
```